src/main/java/org/openstreetmap/atlas/geography/geojson/parser/domain/geometry/MultiPolygon.java

```diff
@@ -1,53 +1,49 @@
 package org.openstreetmap.atlas.geography.geojson.parser.domain.geometry;
 
+import java.util.Collections;
 import java.util.List;
 import java.util.Map;
+import java.util.stream.Collectors;
 
-import org.apache.commons.lang3.Validate;
 import org.openstreetmap.atlas.geography.Polygon;
-import org.openstreetmap.atlas.geography.geojson.parser.domain.bbox.Bbox;
-import org.openstreetmap.atlas.geography.geojson.parser.domain.foreign.ForeignFields;
 import org.openstreetmap.atlas.geography.geojson.parser.domain.geometry.coordinate.Coordinates;
 import org.openstreetmap.atlas.geography.geojson.parser.domain.geometry.coordinate.Position;
 import org.openstreetmap.atlas.geography.geojson.parser.domain.geometry.coordinate.Positions;
+import org.openstreetmap.atlas.utilities.maps.MultiMap;
 
 /**
  * @author Yazad Khambata
  */
 @SuppressWarnings("squid:S2160")
-public class MultiPolygon
-        extends AbstractGeometryWithCoordinateSupport<List<List<Position>>, List<Polygon>>
+public class MultiPolygon extends
+        AbstractGeometryWithCoordinateSupport<List<List<List<Position>>>, org.openstreetmap.atlas.geography.MultiPolygon>
 {
-    private MultiLineString value;
+    private Coordinates<List<List<List<Position>>>> coordinates;
```

Comment:
It seems this needs to be transient or serializable: https://sonarcloud.io/project/issues?id=org.openstreetmap.atlas%3Aatlas&issues=AXTa20XMSXmRf2zUqC-Q&open=AXTa20XMSXmRf2zUqC-Q&pullRequest=686
It might not be an issue with local testing, but if used within Spark it could become one.

Reply:
I am not sure here, but i the object is using that `coordinates` field only at initialization for example, transient might be good enough... Otherwise it will need to be serializable

Reply:
I ended up finding a number of similar classes with the same issue. My solution to make them serializable is not the prettiest, but there is a bunch of clean up around the Coordinates class to really do it right. I think that clean up makes more sense in a separate PR.